### PR TITLE
Enable concurrent kernel execution with ElementwiseKernel and ReductionKernel

### PR DESCRIPTION
**Pull request by koji123**
This PR add "stream" argument to ElementwiseKernel and ReductionKernel to enable concurrent kernel execution with those kernel.

### CODE REVIEW

File: cupy/core/elementwise.pxi [hunk 1]
```diff
@@ -549,7 +550,7 @@ cdef class ElementwiseKernel:
         kern = _get_elementwise_kernel(
             args_info, types, self.params, self.operation,
             self.name, self.preamble, self.kwargs)
-        kern.linear_launch(indexer.size, inout_args)
+        kern.linear_launch(indexer.size, inout_args, shared_mem=0, block_max_size=128, stream=stream)
```
Comment (okuta) [inline cupy/core/elementwise.pxi]: How about `kern.linear_launch(indexer.size, inout_args, stream=stream)`?

Comment (koji123) [inline cupy/core/elementwise.pxi]: Cython does not support this form. https://mail.python.org/pipermail/cython-devel/2013-February/003342.html

Comment (okuta) [inline cupy/core/elementwise.pxi]: Sorry, I did not know that.

Comment (okuta) [inline cupy/core/elementwise.pxi]: Please fix following error. This line is too long.
```./cupy/core/elementwise.pxi:553:80: E501 line too long (101 > 79 characters)```

**Comment by okuta** [comment]
LGTM except comment.

File: cupy/core/elementwise.pxi [hunk 2]
```diff
@@ -549,7 +550,8 @@ cdef class ElementwiseKernel:
         kern = _get_elementwise_kernel(
             args_info, types, self.params, self.operation,
             self.name, self.preamble, self.kwargs)
-        kern.linear_launch(indexer.size, inout_args)
+        kern.linear_launch(indexer.size, inout_args, shared_mem=0,
+            block_max_size=128, stream=stream)
```
Comment (okuta) [inline cupy/core/elementwise.pxi]: Please fix following error.
```./cupy/core/elementwise.pxi:554:13: E128 continuation line under-indented for visual indent```

**Pull request merged**